Add Warning to Display Layer View State Sample (#542)
Esri/arcgis-runtime-samples-java: map_view/display-layer-view-state/README.md

## Use case

The view status includes information on the loading state of layers and whether layers are visible at a given scale. You might change how a layer is displayed in a layer list to communicate whether it is being viewed in the map. For example, you could show a loading spinner next to its name when the view status is `LOADING`, grey out the name when `NOT_VISIBLE` or `OUT_OF_SCALE`, show the name normally when `ACTIVE`, or with a warning or error icon when the status is `WARNING` or `ERROR`.

## How to use the sample

> Click "Load Layer" to add a new layer to the map. The current view status of the layer will display at the top of the sample. Zoom in and out of the map and note that the layer disappears when the map is scaled outside of its min and max scale range. If you disconnect your device from the network and pan around the map, a warning alert will display. Reconnect to the network to remove the warning. Turn the layer's visibility off by unchecking the "layer active" checkbox. The layer's current view status will update accordingly as you carry out these actions.

Reviewer: just some word smithing suggestions:

```suggestion
Click "Load Layer" to add a feature layer to the map. The current view status of the layer will display at the top of the sample. Zoom in and out of the map and note the layer disappears when the map is scaled outside of its min and max scale range. Control the layer's visibility with the "Layer visibility" checkbox. If you disconnect your device from the network and pan around the map, a warning alert will display. Reconnect to the network to remove the warning. The layer's current view status will update accordingly as you carry out these actions.
```

Author: Implemented as above, but making "Layer visible" checkbox as discussed

## Use case

> The view status includes information on the loading state of layers and whether layers are visible at a given scale. You might change how a layer is displayed in a layer list to communicate whether it is being viewed in the map. For example, you could show a loading spinner next to its name when the view status is `LOADING`, grey out the name when `NOT_VISIBLE` or `OUT_OF_SCALE`, show the name normally when `ACTIVE`, or with a warning or error icon when the status is `WARNING` or `ERROR`.

Reviewer 2: I'm a bit puzzled why we're using `view status` in this description, but the sample is called `view state`. Just pointing it out, no need to change anything since this looks like what's in the design.

## How to use the sample

> Click the *Load Layer* button to create a new layer and add it to the map. As you pan and zoom around the map, note how the `LayerViewStatus` flags change; for example, `OUT_OF_SCALE` becomes true when the map is scaled outside of the layer's min and max scale range. Click the *Hide Layer* button to hide the layer and observe the view state change to `NOT_VISIBLE`.

Reviewer: The readme template wiki in common samples shows that this section should be a plain english description of how to run the app, along with "a short statement on what the user will see (technically, as in a raster on a map, etc.) and add a little call to action (i.e. 'pan and zoom to explore')."

This current description (copied from the design) is API terminology heavy, and API terminology should not sit in this section in my experience. I would word smith it to be API term agnostic:

```suggestion
Click "Load Layer" to add a new layer to the map. After it loads, the current view status of the map will display at the top of the sample. Zoom in and out of the map, and note the layer disappears when the map is scaled outside of its min and max scale range. The view status will update accordingly. Turn the layer's visibility off by clicking "Hide Layer". 
```

> If your device supports airplane mode, you can turn this on and pan around the map to see layers display the `WARNING` status when they cannot online fetch data. Turn airplane mode back off to see the warning disappear.

Reviewer 2: I've suggested this previously in the common design:
Perhaps we should use the term `disconnect from the network` instead of `airplane mode`?
After all, airplane mode turns off bluetooth, gps, etc. as well. We don't have to imply that gps or bluetooth have any effect on this.

Either way I think there's a small mixup of words here (@TADraeseke this typo is live in android - sorry, can't seem to tag Hudson):
```suggestion
If your device supports airplane mode, you can turn this on and pan around the map to see layers display the `WARNING` status when they cannot fetch online data. Turn airplane mode back off to see the warning disappear.
```

Reviewer 2: @jenmerritt unless regression testing has destroyed my ability to comprehend text, I think there's still the small issue of the mixed up words here 😄 Just a reminder

Author: Thanks, fixed the word order. I've also amended to disconnect from network, that's really what I was doing when testing, and I'll mention at review.
